cli/sql.go

```diff
@@ -74,11 +75,47 @@ http://www.cockroachlabs.com/docs/
 `)
 }
 
+// handleSet handles the \set command and changes client-side flags.
+func handleSet(cmd []string, confirm *bool) int {
+	if len(cmd) != 2 {
+		fmt.Fprintf(os.Stderr, "Usage:\n \\set <option>\n")
+		return cliNextLine
+	}
+
+	switch cmd[1] {
+	case `confirm`:
+		*confirm = true
+	case `noconfirm`:
+		*confirm = false
+	default:
+		fmt.Fprintf(os.Stderr, "unrecognized \\set option: %s\n", cmd[1])
+	}
+	return cliNextLine
+}
+
+// confirmStatement checks whether the statement string contains
+// "dangerous" statements and asks a confirmation to the user if
+// necessary.
+func confirmStatement(confirm bool, stmt string) bool {
+	stmt = strings.ToUpper(stmt)
+	if confirm && (strings.Contains(stmt, "DROP ")) {
```

Comment:
There can be multiple statements on the same line separated with semicolons.
As explain in my 2nd comment to the issue I think that false positives are acceptable.
They will likely annoy our users less than the added complexity in the code for a narrower check would annoy us in the future.